Assets/Scripts/Enemy/Minions/BeeIdleBehavior.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class beeIdleBehavior : StateMachineBehaviour
+{
+    private BeeController _controller;
+    private Vector2 _playerLocation = Vector2.zero;
+    private Vector3 _startPosition, _endPosition;
+    private float _elapsedTime, _percentageComplete;
+    private float _desiredDuration = 1.5f;
+    private float _angerDistance = 3f;
+    [SerializeField] private AnimationCurve _curve;
+
+    // OnStateEnter is called when a transition starts and the state machine starts to evaluate this state
+    override public void OnStateEnter(Animator animator, AnimatorStateInfo stateInfo, int layerIndex)
+    {
+        _controller = animator.GetComponent<BeeController>();
+        _startPosition = _controller.transform.position;
+        _endPosition = _controller.transform.position;
+    }
+
+    // OnStateUpdate is called on each Update frame between OnStateEnter and OnStateExit callbacks
+    override public void OnStateUpdate(Animator animator, AnimatorStateInfo stateInfo, int layerIndex)
+    {
+        
+        // Check if the bee should be angry
+        _playerLocation = GameObject.FindWithTag("Player").transform.position;
```

Comment:
Store the player object in a var in onStateEnter and then just grab its position here, so that you don't have to search for the player object every frame.